sympy/physics/units/util.py

```diff
@@ -161,6 +161,15 @@ def check_dimensions(expr, unit_system="SI"):
     from sympy.physics.units import UnitSystem
     unit_system = UnitSystem.get_unit_system(unit_system)
 
+    def addDict(dict1, dict2):
+        """Merge dictionaries by adding values of common keys and
+        removing keys with value of 0."""
+        dict3 = {**dict1, **dict2}
+        for key, value in dict3.items():
+            if key in dict1 and key in dict2:
+                   dict3[key] = value + dict1[key]
+        return {key:val for key, val in dict3.items() if val != 0}
```

Comment:
I was about to suggest using `collections.Counter`, which is a subclass of `dict` for counting, but apparently it doesn't work with negative countings.